kusto/kcsb.go

```diff
@@ -251,6 +254,15 @@ func (kcsb *ConnectionStringBuilder) WithSystemManagedIdentity() *ConnectionStri
 	return kcsb
 }
 
+// WithKubernetesWorkloadIdentity Creates a Kusto Connection string builder that will authenticate with AAD application, using
+// an application token obtained from a Microsoft Service Identity endpoint using Kubernetes workload identity.
+func (kcsb *ConnectionStringBuilder) WithKubernetesWorkloadIdentity() *ConnectionStringBuilder {
```

Comment:
The API ought to take in any argument that is required to configure this auth mode. 
This one seems to require 3 (app id, token file, authority id)

Reply:
This ought to also affect tests and code example.

Reply:
done